### ISSUE TITLE
Implement message dump method for bufr

### ISSUE THREAD

**Issue by sandorkertesz** [ecmwf/earthkit-data#115]
It would be useful if there was a `dump` method available on a BUFR data object displaying the contents of the given message . In a jupyter notebook it should display this information as an interactive tree view showing the hierarchy of data (just like in `bufr_dump` or `codes_ui`). 

Since the bufr object does not have a slice operator/message iterator at the moment, the first argument to `dump` should be the message index. E.g. to dump the second message we could use:

```python
import earthkit.data
ds = erarhkit.data.from_source("file", "temp.bufr")
ds.dump(1)
```

The goal is to develop a pure html/css tree view that can be used in a Jupyter notebook without needing to install ipywidgets.

**Comment by sandorkertesz** [comment]
Unfortunately, it does not work.  Use ecCodes Version 2.30.2.

<img width="808" alt="image" src="https://github.com/ecmwf/earthkit-data/assets/9033020/e8ee9da9-1b4a-4ba4-a8fe-f822f46493de">

**Comment by sandorkertesz** [comment]
Fixed by #124
